### ISSUE TITLE
Facebook share count always zero?

### ISSUE THREAD

**Issue by uwej711** [heiseonline/shariff-backend-php#13]
We currently get always zeros for facebook share counts, when using the method the facebook backend uses. If I use the graph api id method (which needs an access token to work, I get the actual share count. 

Anybody else observed something similar? I can try to do a PR but the method mentioned above needs an facebook appId and secret that needs to be configured.

https://graph.facebook.com/v2.2/?id=http://www.heise.de/&access_token=...
```
{
   "og_object": {
      "id": "419033995599",
      "description": "News und Foren zu Computer, IT, Wissenschaft, Medien und Politik. Preisvergleich von Hardware und Software sowie Downloads beim Heise Zeitschriften Verlag.",
      "title": "IT-News, c't, iX, Technology Review, Telepolis",
      "type": "website",
      "updated_time": "2015-01-11T17:55:57+0000",
      "url": "http://www.heise.de/"
   },
   "share": {
      "comment_count": 0,
      "share_count": 9445
   },
   "id": "http://www.heise.de/"
}
```

https://graph.facebook.com/fql?q=SELECT%20total_count%20FROM%20link_stat%20WHERE%20url=%22http://www.heise.de%22
```
{
   "data": [
      {
         "total_count": 5565
      }
   ]
}
```

**Comment by uwej711** [comment]
OK, looks good now, I was not patient enough. 
But fql could be deprecated sometime,  so switching to the graph api could be necessary some day.

**Comment by devimplode** [comment]
Nope, you are right with the FQL method. And no, it doesn't matter if you are patient.
As I already partly explained in Issue #3, there are some huge problems with the different APIs.
I also noticed in the last days that the `/v1/fql` method didn't use the actual values for share count. It seems it serves only cached ones and didn't update it's cache.
The graph method however serves the right values and updates the cache of the fql api.
Also, yes - the `/v1/fql`, `/v2.0/fql`, rest, and graph-`/v1/`-api are deprecated.

As you mentioned, the Graph-API, to be specific `>= /v2.0/`, needs a api key - witch everyone can easily create. (Unlike crappy api's like pinterest)
But please don't think that this is a perfect solution!
For example look at your own code:
- FQL: 5565
- Graph`v2.2`: 9445
- Graph`v1`: 19
- Offic.Button: 5585

The Graph-API `/v1/` on the other hand is deprecated, doesn't need a api key and is *really strange* :tm:.
In our test example `http://www.heise.de/` it didn't serves any like count (`19` doesn't count!)
For other urls it works better.

#### Recommendation
- Set up config variables for `FB_AppID` / `FB_AppSecret`.
- In the Facbook-Backend try the Graph `/v2.2/` method if credentials are working
- If the admin doesn't like facebook / didn't want an api key - you know we build this project because we doesn't like such things
 - fall back to fql - method - as long it works

Oh and please reopen this ticket - this could be a long road...

Have a nice weekend! ;)

**Comment by uwej711** [comment]
I also noticed the different numbers, tried it with other urls that display a button and got three different numbers. On our site I finally got the right count after waiting a while. Maybe anybody has more insight which method returns which number. Maybe make the method configurable here ...

**Comment by uwej711** [comment]
Just looked at issue #3, so while playing around I called the graph method which resulted in the FQL returning the correct value ... I thought it was just a question of waiting ...

**Comment by uwej711** [comment]
Closed via #16
